airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/VersionedAirbyteMessageBufferedWriterFactory.java

```diff
@@ -25,6 +29,11 @@ public VersionedAirbyteMessageBufferedWriterFactory(final AirbyteMessageSerDePro
 
   @Override
   public AirbyteMessageBufferedWriter createWriter(BufferedWriter bufferedWriter) {
+    final boolean needMigration = !protocolVersion.getMajorVersion().equals(migratorFactory.getMostRecentVersion().getMajorVersion());
+    LOGGER.info(
```

Comment:
This class is a factory so I am assuming that it is called only once per sync and not once per message. Could you confirm that? If not, we need to switch this log to debug to ensure that we won't generate gigantic logs.

Reply:
Yes, this is called once per sync.